python/cvmfs/repository.py

```diff
@@ -198,38 +205,223 @@ def _pop_catalog(self):
 
 
 
-class Repository:
-    """ Abstract Wrapper around a CVMFS Repository representation """
+class Cache:
+    def __init__(self, cache_dir):
+        if not os.path.exists(cache_dir):
+            cache_dir = tempfile.mkdtemp(dir='/tmp', prefix='cache.')
+        self._cache_dir = cache_dir
+        self._create_cache_structure()
+        self._cleanup_metadata()
+
+    def _cleanup_metadata(self):
+        for f in glob.glob(self._cache_dir + '/' + '.cvmfs*'):
+            os.remove(f)
+
+
+    def _create_dir(self, path):
+        cache_full_path = os.path.join(self._cache_dir, path)
+        if not os.path.exists(cache_full_path):
+            os.mkdir(cache_full_path, 0755)
+
+
+    def _create_cache_structure(self):
+        self._create_dir('data')
+        for i in range(0x00, 0xff + 1):
+            new_folder = '{0:#0{1}x}'.format(i, 4)[2:]
+            self._create_dir(os.path.join('data', new_folder))
+
+
+    def cache_path(self):
+        return str(self._cache_dir)
+
+
+    def add(self, file_name):
+        full_path = os.path.join(self._cache_dir, file_name)
+        return open(full_path, 'w+')
+
+
+    def get(self, file_name):
+        full_path = os.path.join(self._cache_dir, file_name)
+        if os.path.exists(full_path):
+            return open(full_path, 'rb')
+        return None
+
+
+    def evict(self):
+        data_path = os.path.join(self._cache_dir, 'data')
+        shutil.rmtree(data_path)
+        self._create_cache_structure()
+
+
+
+class Fetcher:
```

Comment:
Maybe this article makes my point clearer: https://en.wikipedia.org/wiki/Is-a
Inheritance should happen when an sub-entity can be considered a member of a top-level entity. For that, one has to ask himself if that sub-entity meets the relationship IS-A with the parent-entity. For example: a human IS-An animal, so human could inherit from animal.

But is a fetcher a cache? I don't think so.

Just as an absurd example that came up to my mind: it's like if you plan to make a bus driver inherit from bus and human at the same time. No, a bus driver has an attribute, which is his vehicle :P